### ISSUE TITLE
Use the Microsoft.Storage resource provider for Azure file share management operations

### ISSUE THREAD

**Issue by wmgries** [Azure/azure-cli#11583]
**Resource Provider**
Microsoft.Storage

**Description of Feature or Work Requested**
Today, the Azure Files CLI commands for file share management (az storage share) go through the Azure storage data plane. This causes a number of issues for customers, most notably it means customers are unable to manage their file shares if their Azure storage account is in a vnet and the client where they are running the Azure CLI (such as their laptop) isn't - this is a very common management scenario for Azure Files. Therefore, we would like to add a new command path, az storage share-rm, that sends the file share management requests through the Storage Resource Provider (SRP) instead of through the Azure storage data plane. 

We need to add a new command path for several reasons. First and most importantly, not all operations are available on the file share proxy resource object offered by SRP, by design. For example, operations to create and manage SAS tokens are not planned and are unlikely to be ever added to the file share object. Secondly, SRP is being a middle-man for file share requests - the requests they get to manage a file share are passed through to the data plane APIs (SRP has a way through customer vnets). SRP is generally believed to have less performance than the underlying data plane API. Therefore, it is strongly recommended that customers needing to do many calls against the Azure file share at once should use the data plane. Although this is not the primary thing that customers use Azure Files for today, we have to maintain this path for compatibility reasons. Finally, adopting SRP will require some significant changes to the command interface, and we would like to avoid a breaking change. Note, this change has already been done in the Azure PowerShell package - we maintain (*verb*-Az*noun*, which goes through the data plane, and *verb*-AzRm*noun*, which goes through SRP).

**Minimum API Version Required**
Microsoft.Storage 2019-06-01

**Swagger Link**
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/file.json

**Target Date**
1/13

Here's the link to the [OneNote describing the change](https://microsoft.sharepoint.com/teams/AzureStorage/_layouts/OneNote.aspx?id=%2Fteams%2FAzureStorage%2FShared%20Documents%2FXFiles%2FXFile%20Development&wd=target%28Feature%20Areas%2FAPI.one%7C536266A2-908A-4781-B342-6EDA5B6528CA%2FCLI%20Control%20Plane%7C0EB3A489-DFAF-4469-8EA5-94336737DAA5%2F%29).

**Comment by yonzhan** [comment]
@Juliehzl  add to S164.

**Comment by jsntcy** [comment]
@wmgries, in your description, you mentioned that "Minimum API Version Required
Microsoft.Storage 2019-06-01", is this feature only supported in 2019-06-01 as I checked https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2019-04-01/blob.json and found the API spec of 2019-04-01 is very similar with 2019-06-01?

**Comment by yonzhan** [comment]
@wmgries, could you please help to confirm the question mentioned above?

**Comment by wmgries** [comment]
Hey, sorry I missed this. I don't see why you cannot use 2019-04-01, however, the control plane is missing several features currently, in both API versions, that are in plan to be added in January. Several examples: snapshot support and share stats. These will presumably use 2019-06-01 as I do not think they require breaking changes to the API. 

The spec I gave does not assume this missing features are there.

**Comment by jsntcy** [comment]
@wmgries , thanks for your confirmation. We'll use 2019-04-01 for this feature and in the future, when we support features only in 2019-06-01, we'll use 2019-06-01.

Also this time for each command, CLI will not support the following options as PowerShell didn't support them yet. If we do need support them, we can support them in the future.
- We only support using "account-name", but not support "account-id".
- We will not support "ids" for the commands in spec, such as "az storage share-rm show --ids string[]>"

**Comment by wmgries** [comment]
I would argue the equivalent to `account-id` for PowerShell is taking in an object. For example:

```PowerShell
$storAcct = Get-AzStorageAccount -ResourceGroupName "myRG" -Name "mysa"
New-AzRmStorageShare -StorageAccount $storAcct -Name "myshare"
```

CLI does let you get storage accounts by the ID of the storage account, (`az storage account show --ids`), so if you're in the position where you have the ID of a storage account, I don't see why you shouldn't be able to create a proxy resource with it instead of having to independently lookup the name and resource group of the resource.

**Comment by jsntcy** [comment]
@wmgries, thanks for your comments. We'll support account-id as well.
But for the commands (exists, delete, show, update), we'll not support option "--ids" to keep consistent with PowerShell and data plane file share commands as neither PowerShell nor data plane file share support "--ids", does this decision make sense to you?

**Comment by jsntcy** [comment]
This feature was already released in [Azure CLI 2.0.81](https://docs.microsoft.com/en-us/cli/azure/storage/share-rm?view=azure-cli-latest).
